src/index.ts

```diff
@@ -1,18 +1,106 @@
 import base64 = require('base-64');
+import queryStringHelpers = require('querystring');
+import urlHelpers = require('url');
+
+const emptyStringToUndefined = (str: string): string | undefined => (str === '' ? undefined : str);
+
+// TS doesn't return the correct type for array and object index signatures. It returns `T` instead
+// of `T | undefined`. These helpers give us the correct type.
```

Comment:
If you're interested, I've proposed a compiler option to fix this: https://github.com/Microsoft/TypeScript/issues/13778

Give it a vote ;-) /cc @samijaber